bolt-sidecar/src/api/commitments/firewall/receiver.rs

```diff
@@ -0,0 +1,564 @@
+                            url.clone(),
+                            self.chain,
+                            None,
+                            signer.clone(),
+                        )
+                        .to_jwt()
+                        .expect("failed to produce JWT");
+
+                        let api_events_tx = api_events_tx.clone();
+                        let ping_rx = ping_rx.resubscribe();
+                        let shutdown_rx = shutdown_rx.resubscribe();
+
+                        async move {
+                            handle_connection(url, state, jwt, api_events_tx, ping_rx, shutdown_rx)
+                                .await
+                        }
+                    },
+                    |err| {
+                        let is_shutdown = matches!(
+                            err,
+                            ConnectionHandlerError::ProcessorInterrupted(InterruptReason::Shutdown)
+                        );
+                        if is_shutdown {
+                            info!("Shutting down websocket connection");
+                        } else {
+                            error!(?err, "error while handling websocket connection. Retrying...");
+                        }
+
+                        !is_shutdown
+                    },
+                )
+                .await
+            });
+        }
+
+        api_events_rx
+    }
+
+    /// Returns the local addr the server is listening on (or configured with).
+    pub fn urls(&self) -> &[Url] {
+        self.urls.as_slice()
+    }
+}
+
+async fn handle_connection(
+    url: String,
+    state: ProcessorState,
+    jwt: String,
+    api_events_tx: mpsc::Sender<CommitmentEvent>,
+    ping_rx: broadcast::Receiver<()>,
+    shutdown_rx: broadcast::Receiver<()>,
+) -> Result<(), ConnectionHandlerError> {
+    let ws_config =
+        WebSocketConfig { max_message_size: Some(MAX_MESSAGE_SIZE), ..Default::default() };
+
+    let mut request = url.clone().into_client_request()?;
+    request
+        .headers_mut()
+        .insert("Authorization", format!("Bearer {}", jwt).parse().expect("valid header"));
+
+    match connect_async_with_config(request, Some(ws_config), USE_NAGLE).await {
+        Ok((stream, response)) => {
+            info!(?url, ?response, "opened websocket connection");
+            let (write_sink, read_stream) = stream.split();
+
+            // For each opened connection, create a new commitment processor
+            // able to handle incoming message requests.
+            let commitment_request_processor = CommitmentRequestProcessor::new(
+                url,
+                state,
+                api_events_tx.clone(),
+                write_sink,
+                read_stream,
+                ping_rx.resubscribe(),
+                shutdown_rx.resubscribe(),
+            );
+            // Run the commitment processor indefinitely, reconnecting on failure.
```

Comment:
I don't understand how this reconnection works. Could you please elaborate?

Reply:
could be handled in a future PR, this one is already huge, but I would make sure reconnection with exponential backoff is present